### PR TITLE
Scaffold/screens

### PR DESCRIPTION
**Pull request by barun1997**
Add product, category and order listing screens
Add product and category form screens
Integrate with firebase

### CODE REVIEW

**Comment by npsedhain** [comment]
Sir ali sano sano pr kholirakham hai baru :P

File: src/constants/colorByStatus.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Colors } from 'react-native-paper';
+
+type colorByStatusType = {
+	[key: string]: string;
+};
+
+const colorByStatus: colorByStatusType = {
```
Comment (npsedhain) [inline src/constants/colorByStatus.ts:8]: Sir yo 0, 1, 2 rakhnu ko kei specific reason ?

File: src/constants/fieldIncrementByOne.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { firebase } from '@react-native-firebase/firestore';
```
Comment (npsedhain) [inline src/constants/fieldIncrementByOne.ts]: This file lai firebase util bhanera utils/firebase/.... esma rakhda thik hola

File: src/schemas/Product.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import * as Yup from 'yup';
+
+const SUPPORTED_FORMATS = ['image/jpg', 'image/jpeg', 'image/gif', 'image/png'];
```
Comment (npsedhain) [inline src/schemas/Product.ts]: Esto arrays harulai constants bhitra rakhda better hai

File: src/screens/Categories/screens/AddCategory/AddCategory.tsx [hunk 4]
```diff
@@ -1,11 +1,10 @@
 import { Formik } from 'formik';
 import React from 'react';
-import AddCategoryForm from './components/AddCategoryForm';
+import CategoryForm from './components/CategoryForm';
 import { addCategory } from '../../../../services/categoryService';
 import { useNavigation } from '@react-navigation/native';
```
Comment (npsedhain) [inline src/screens/Categories/screens/AddCategory/AddCategory.tsx]: better practice chai package imports haru top ma rakhne ani module imports haru tala ani tyo ni ladder structure bhaye best. 
Ani constants ko euta section, components ko euta, utils ko euta teseri, ek ek line ko gap for all

File: src/screens/Categories/screens/AddCategory/components/CategoryForm.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+import { Button, TextInput, useTheme } from 'react-native-paper';
+import { CategoryType } from '../../../../../interfaces/Category';
+import FormProps from '../../../../../interfaces/FormProps';
+import { widthPercentageToDP as wp } from 'react-native-responsive-screen';
+const CategoryForm: React.FC<FormProps<CategoryType>> = ({
```
Comment (npsedhain) [inline src/screens/Categories/screens/AddCategory/components/CategoryForm.tsx:8]: Hya ek line spacing bhaako ramro

File: src/screens/Products/screens/AddProduct/AddProduct.tsx [hunk 6]
```diff
@@ -1,24 +1,62 @@
...
 	description: '',
 	category: '',
 };
 
 function AddProductScreen(): JSX.Element {
+	const navigation = useNavigation();
+
+	const handleSubmit = async (values: ProductType): Promise<void> => {
+		try {
+			console.log(values);
```
Comment (npsedhain) [inline src/screens/Products/screens/AddProduct/AddProduct.tsx:24]: linter add garne ho ki ? :P

File: src/services/categoryService.ts [hunk 7]
```diff
@@ -1,30 +1,35 @@
 import firestore from '@react-native-firebase/firestore';
-interface CategoryType {
-	name: string;
-}
+import { fieldIncrementByOne } from '../constants/fieldIncrementByOne';
+import { CategoryType } from '../interfaces/Category';
 
+//TODO: Implement for individual stores
```
Comment (npsedhain) [inline src/services/categoryService.ts:5]: 👍

File: src/utils/statusToString.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable indent */
+function statusToString(statusNumber: number): string {
```
Comment (npsedhain) [inline src/utils/statusToString.ts]: baru number nai use nagarne katai sidhai constants mai pending haru halna milne thena ?

Comment (npsedhain) [reply src/utils/statusToString.ts]: Oi yesma esto garnu sato sidhai enum haldey na

Comment (barun1997) [reply src/utils/statusToString.ts]: Firebase bata enum aucha ra? Maile online khojeko ani audaina bhanya jasto bhayera esto gareko. Ki taile k garne bhaneko?  @npsedhain

Comment (barun1997) [reply src/utils/statusToString.ts]: @npsedhain  I think taile bhaneko yo nai ho. latest commit her ta dai

**Pull request merged**